src/attribute_type.rs

```diff
@@ -113,12 +120,16 @@ pub enum SdpAttributeCandidateTransport {
     Tcp,
 }
 
-impl ToString for SdpAttributeCandidateTransport {
-    fn to_string(&self) -> String {
-        match *self {
-            SdpAttributeCandidateTransport::Udp => "UDP".to_string(),
-            SdpAttributeCandidateTransport::Tcp => "TCP".to_string(),
-        }
+impl fmt::Display for SdpAttributeCandidateTransport {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
```

Comment:
There is a comment in the next file about this pattern. I didn't realize that github hid large diffs during review.

```rust
match *self {
    SdpAttributeCandidateTransport::Udp => "UDP",
    SdpAttributeCandidateTransport::Tcp => "TCP",
}.fmt(f)
```